src/org/apache/storm/metrics2/RocksConnector.java

```diff
@@ -22,51 +22,76 @@
 import java.util.List;
 import java.util.ArrayList;
 import java.util.HashMap;
+import java.io.File;
 
 import org.rocksdb.RocksDB;
 import org.rocksdb.Options;
 import org.rocksdb.RocksDBException;
 import org.rocksdb.RocksIterator;
 
+/**
+ * This class implements the Storm Metrics Store Interface using RocksDB as a store
+ * It contains preparing, insertion and scan methods to store and query metrics
+ *
+ * @author Austin Chung <achung13@illinois.edu>
+ * @author Abhishek Deep Nigam <adn5327@gmail.com>
+ * @author Naren Dasan <naren@narendasan.com>
+ */
 
-public class RocksConnector {
+public class RocksConnector implements MetricStore {
 
     private RocksDB db;
-    public RocksConnector(String dbfilename)
-    {
+
+    /**
+     * Implements the prepare method of the Metric Store, create RocksDB instance
+     * using the configurations provided via the config map
+     * @param config Storm config map
+     */
+    @Override
+    public void prepare(HashMap config) {
+
+        try {
+            validateConfig(config);
+        } catch(MetricException e) {
+            System.out.println(e);
+        }
         RocksDB.loadLibrary();
         // the Options class contains a set of configurable DB options
         // that determines the behavior of a database.
-        Options options = new Options().setCreateIfMissing(true);
+        boolean createIfMissing = Boolean.parseBoolean(config.get("create-if-missing").toString());
+        Options options = new Options().setCreateIfMissing(createIfMissing);
+
         this.db = null;
         try {
             // a factory method that returns a RocksDB instance
-            this.db = RocksDB.open(options, dbfilename);
+            String path = config.get("store-location").toString();
```

Comment:
the configs should be namespaced: storm.metrics.store.rocksdb... so storm.metrics.store.rocksdb.location in this case